adorym/ptychography.py

```diff
@@ -393,37 +401,38 @@ def reconstruct_ptychography(
             opt = optimizer
             opt.name = 'obj'
         else:
+            opt = get_optimizer(optimizer)
             if optimizer == 'adam':
                 optimizer_options_obj = {'step_size': learning_rate}
```

Comment:
I think the following lines are not ideal - I don't think anymore that every optimizer should be the same like this.